x-pack/platform/plugins/shared/ml/public/application/model_management/deployment_params_mapper.ts

```diff
@@ -93,10 +96,16 @@ export class DeploymentParamsMapper {
      */
     const maxSingleMlNodeProcessors = this.mlServerLimits.max_single_ml_node_processors;
 
-    this.threadingParamsValues = new Array(THREADS_MAX_EXPONENT)
-      .fill(null)
-      .map((v, i) => Math.pow(2, i))
-      .filter(maxSingleMlNodeProcessors ? (v) => v <= maxSingleMlNodeProcessors : (v) => true);
+    this.threadingParamsValues = Array.from(
+      /** To allow maximum number of threads to be 2^THREADS_MAX_EXPONENT */
+      {
+        length:
+          Math.floor(
+            Math.log2(this.showNodeInfo ? MAX_NUMBER_OF_THREADS : MAX_NUMBER_OF_THREADS_SERVERLESS)
```

Comment:
`showNodeInfo` is basically being used here as an `isServerless` in this file. None of the checks it's being used for are about whether or not to show not information.

Maybe we should change it to a `useReducedThreads` or something else more vague.

Or maybe this one file should be allowed to have a `isServerless` flag as a lot of different logic is depending or whether or not we're in serverless.

Reply:
I refactored it to completely get rid of the `showNodeInfo` [here](https://github.com/elastic/kibana/pull/227940/commits/32bf77d2791292366a8ef393650c2a5b433dcc9a).

Now everything is retrieved from the configuration file.